lib/resources/storage_provider.dart

```diff
@@ -1,11 +1,12 @@
-import 'package:cached_network_image/cached_network_image.dart';
-import 'package:firebase_storage/firebase_storage.dart';
+import 'package:acs_upb_mobile/resources/storage/fire_storage_service.dart';
 import 'package:flutter/material.dart';
 
 class StorageProvider with ChangeNotifier {
-  Future<ImageProvider<dynamic>> imageFromPath(String path) async {
-    StorageReference ref = FirebaseStorage.instance.ref().child(path);
-    String url = await ref.getDownloadURL();
-    return CachedNetworkImageProvider(url);
+  Future<String> findIconUrl(BuildContext context, String iconPath) async {
```

Comment:
I'd call this `image` instead of `icon`. True, we're currently using it for the website icon, but we'll also use it for the profile picture and other things, so `image` is more generic.